src/orders/components/OrderRefundReturnAmount/OrderRefundReturnAmount.tsx

```diff
@@ -200,8 +201,29 @@ const OrderRefundAmount: React.FC<OrderRefundAmountProps> = props => {
       ? refundTotalAmount?.amount
       : paymentsTotalAmount?.amount;
 
-  const isAmountTooSmall = selectedRefundAmount && selectedRefundAmount <= 0;
-  const isAmountTooBig = selectedRefundAmount > maxRefund?.amount;
+  const isAnyPaymentAmountTooSmall =
+    order?.payments.filter(payment => {
+      const currentPayment = data.paymentsToRefund?.find(getById(payment.id));
```

Comment:
Since logic on lines 206 and 215 is the same you could create a function out of it, something like `getCurrentPaymentValue` and just pass `payment.id` as a param.